core/common/src/main/java/alluxio/cli/CommandUtils.java

```diff
@@ -120,16 +117,16 @@ public static void checkNumOfArgsNoMoreThan(Command cmd, CommandLine cl, int n)
    * @return list of the node names, null when file fails to read
    */
   @Nullable
-  public static List<String> readNodeList(String confDir, String fileName) {
+  public static Set<String> readNodeList(String confDir, String fileName) {
     List<String> lines;
     try {
       lines = Files.readAllLines(Paths.get(confDir, fileName), StandardCharsets.UTF_8);
     } catch (IOException e) {
       System.err.format("Failed to read file %s/%s. Ignored.", confDir, fileName);
-      return new ArrayList<>();
+      return new HashSet<>();
```

Comment:
This can be a Set as we mostly just put its results into a set

Reply:
Not at all. This method is only used to read `conf/masters` and `conf/workers` files. It's fine to put duplicated hostnames there, the duplicates are just ignored. I believe if you chase down the usage of this method the lists all end up being put into sets.

Reply:
It would be nice to warn user of duplicates. We do iterate through all non-comment lines in some bash scripts.

Reply:
Added printouts to System.out.